client/skills.go

```diff
@@ -0,0 +1,146 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package client
+
+import (
+	"bytes"
+	"encoding/json"
+)
+
+// Skill represents a capacity offered by a snap.
+type Skill struct {
+	Name  string                 `json:"name"`
+	Snap  string                 `json:"snap"`
+	Type  string                 `json:"type,omitempty"`
+	Attrs map[string]interface{} `json:"attrs,omitempty"`
+	Apps  []string               `json:"apps,omitempty"`
+	Label string                 `json:"label,omitempty"`
+}
+
+// Slot represents the potential of a given snap to use a skill.
+type Slot struct {
```

Comment:
Let's please not do that yet, per conversation online. This would be just saving a few lines, rather than being semantically correct (a slot isn't just a skill). These will likely diverge soon, and if they don't we'll need to do that via a third private type so we hold the option to diverge them without breaking code.